.github/workflows/codeql-analysis.yml

```diff
@@ -0,0 +1,44 @@
+name: "CodeQL"
+
+on:
+  push:
+    branches: [ "master" ]
+  pull_request:
+    # The branches below must be a subset of the branches above
+    branches: [ "master" ]
+    # Skip the workflow if PR only contains changes to files matching the following path patterns
+    paths-ignore:
```

Comment:
Should this also skip `tests.py` because that's ignored in `codeql-config.yml`?

Reply:
Good catch. Yes, we should.